iidm/iidm-api/src/main/java/com/powsybl/iidm/network/DanglingLine.java

```diff
@@ -277,9 +285,15 @@ default Generation getGeneration() {
     String getUcteXnodeCode();
 
     default Boundary getBoundary() {
```

Comment:
@zamarrenolm and @miovd I think this warning is not acceptable. I think that we have to force the implementation with no default behaviour.

Reply:
I am ok with removing the default implementation. Maybe it was kept to avoid a breaking change.

This fixes a problem not directly related to the purpose of the PR.

The errors that may be caused by keeping the old default implementation (creating a new DL impl everytime the method is called) may be difficult to trace back to this cause. 

It was first observed in the merging view: boundaries of lower level IGMs are mapped to its corresponding adapters in the merging view network, but everytime the merging view is asked for a dangling line boundary, it will ask its lower level IGM for its dangling line boundary to find its adapter. Because the default implementation returned a new dangling line impl for each request, the index of the merging view could not find its own adapter.